zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -531,60 +532,53 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         };
 
         self.tx_id
-            .assign(region, offset, Value::known(F::from(tx.id as u64)))?;
+            .assign(region, offset, Value::known(F::from(tx_id)))?;
         self.tx_nonce
-            .assign(region, offset, Value::known(F::from(tx.nonce)))?;
+            .assign(region, offset, Value::known(F::from(tx.nonce.as_u64())))?;
         self.tx_gas
-            .assign(region, offset, Value::known(F::from(tx.gas)))?;
+            .assign(region, offset, Value::known(F::from(tx.gas())))?;
         self.tx_gas_price
             .assign_u256(region, offset, tx.gas_price)?;
         self.mul_gas_fee_by_gas
-            .assign(region, offset, tx.gas_price, tx.gas, gas_fee)?;
+            .assign(region, offset, tx.gas_price, tx.gas(), gas_fee)?;
         self.tx_caller_address
-            .assign_h160(region, offset, tx.caller_address)?;
+            .assign_h160(region, offset, tx.from)?;
         self.tx_caller_address_is_zero.assign_u256(
             region,
             offset,
-            U256::from_big_endian(&tx.caller_address.to_fixed_bytes()),
+            U256::from_big_endian(&tx.from.to_fixed_bytes()),
         )?;
         self.tx_callee_address
-            .assign_h160(region, offset, tx.callee_address)?;
-        self.call_callee_address.assign_h160(
-            region,
-            offset,
-            if tx.is_create {
-                get_contract_address(tx.caller_address, tx.nonce)
-            } else {
-                tx.callee_address
-            },
-        )?;
+            .assign_h160(region, offset, tx.to_or_contract_addr())?;
+        self.call_callee_address
+            .assign_h160(region, offset, tx.to_or_contract_addr())?;
```

Comment:
Wait, I realized I confused tx_callee_address and call_callee_address. 
I need to figure out what's the difference here.

Reply:
I checked the deleted `zkevm-circuits/src/witness/tx.rs` and confirmed that `tx_callee_address` and `call_callee_address` are currently assigned with the same value.